test/function.c

```diff
@@ -1,12 +1,12 @@
 /**
  * Foo function.
  */
-int foo(int bar, int baz);
+static inline int foo(int bar, int baz);
 
 /**
  * No parameters.
  */
-int no_parameters(void);
+extern int no_parameters(void);
```

Comment:
Looking at the implementation, I'm not quite sure how we get `extern` for functions, but the same applies here as for variables. I'd prefer not having them. (Moreover, extern is superfluous for function declarations in C.)

I think `static` is useful. (Although documenting static functions would be more like documenting implementation details than an API, but it's still fine.)

I think `static inline` should mostly be an implementation detail of an API, but gut feeling says to err on the side of having it than trying to hide it, especially if we're going to have `static` anyway.

Reply:
> Looking at the implementation, I'm not quite sure how we get `extern` for functions, but the same applies here as for variables. I'd prefer not having them. (Moreover, extern is superfluous for function declarations in C.)

We don't, I put it here to make it obvious that it doesn't show up in the documentation in this case. I agree, it's totally meaningless for functions unlike for variables.

> I think `static` is useful. (Although documenting static functions would be more like documenting implementation details than an API, but it's still fine.)
> 
> I think `static inline` should mostly be an implementation detail of an API, but gut feeling says to err on the side of having it than trying to hide it, especially if we're going to have `static` anyway.

Hmm, I think it conveys _something_ as an API, depending on user's expectations. Maybe some of this stuff could be made configurable one day if someone feels strongly about it.